jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.jdi.light.material.elements.utils;
+
+import com.epam.jdi.light.common.JDIAction;
+import com.epam.jdi.light.elements.base.UIBaseElement;
+import com.epam.jdi.light.elements.complex.ISetup;
+import com.epam.jdi.light.elements.interfaces.base.HasClick;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.MarkupLocator;
+import com.epam.jdi.light.material.annotations.JDIModal;
+import com.epam.jdi.light.material.annotations.JDISnackbar;
+import com.epam.jdi.light.material.asserts.utils.ModalAssert;
+import com.epam.jdi.light.material.elements.feedback.Snackbar;
+import org.openqa.selenium.By;
+
+import java.lang.reflect.Field;
+
+import static com.epam.jdi.light.elements.pageobjects.annotations.objects.FillFromAnnotationRules.fieldHasAnnotation;
+
+public class Modal extends UIBaseElement<ModalAssert> implements HasClick, ISetup {
+
+    String modalTitle;
+    String modalDescription;
+    String modalInnerButton;
+    String innerModalPage;
+
+    @JDIAction("Is '{name}' enabled")
+    public void getModalButton() {
+        core().click();
+    }
+
+    @JDIAction("Is '{name}' opened")
+    public boolean isInnerModalOpened() {
+        return core().hasClass(innerModalPage);
+    }
+
+    @JDIAction("Get '{name}'")
+    public boolean verifyModalTitle(String actualTitle) {
+        return core().find(modalTitle).getText().equals(actualTitle);
+    }
+
+    @JDIAction("Get '{name}'")
+    public boolean verifyModalDescription(String actualDescription) {
+        return core().find(modalDescription).getText().equals(actualDescription);
+    }
+
+    @JDIAction("Get '{name}'")
+    public void getInnerModalWindow() {
+        core().find(modalInnerButton);
+    }
+
```

Comment:
Please add:

  @Override
  public ModalAssert is() {
    return new ModalAssert().set(this);
  }